src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -1061,54 +1063,111 @@ void WindowImplX11::setVideoMode(const VideoMode& mode)
     if (mode == VideoMode::getDesktopMode())
         return;
 
-    // Check if the XRandR extension is present
-    int version;
-    if (!XQueryExtension(m_display, "RANDR", &version, &version, &version))
+    // Check if the XRandR extension
+    int xRandRMajor, xRandRMinor;
+    if (!checkXRandR(xRandRMajor, xRandRMinor))
```

Comment:
As strange as it sounds, I've experienced implementations where not calling `XQueryExtension` would cause the extension to fail even if it should be supported. It seemed like calling `XQueryExtension` would load the extension or something. So even if it should be supported, performing the runtime "check" is still the safer variant.